include/benchmark/benchmark.h

```diff
@@ -565,17 +565,21 @@ class State {
   int range_y() const { return range(1); }
 
   BENCHMARK_ALWAYS_INLINE
-  size_t iterations() const { return (max_iterations - total_iterations_) + 1; }
+  int64_t iterations() const {
+    return static_cast<int64_t>(
+      static_cast<uint64_t>(
+        (max_iterations - total_iterations_) + 1));
+  }
 
  private:
   bool started_;
   bool finished_;
-  size_t total_iterations_;
+  uint32_t total_iterations_;
```

Comment:
i buy your 'int' argument and if we need a guarantee, which we do, then this should be ?int32_t. The Google C++ style guide, interestingly, states that we can assume an int is [at least 32 bits](https://google.github.io/styleguide/cppguide.html#Integer_Types), which is a bit surprising.

However, in the same section it also specifically recommends avoiding unsigned unless we need overflow modulo 2^N or representing bit patterns. Given we're doing neither, this should be signed.

So: `int32_t` please. And then promoting once to int64_t in 'iterations' seems reasonable, though i'd promote the individual parts of the calculation. ie, `max_iterations - total_iterations_` could be INT_MAX already if total is 0. the +1 will cause an overflow before you get the chance to promote to int64.

Reply:
I'm sorry, I don't buy the `int` vs. `int32_t` argument.

According to [cppreference](http://en.cppreference.com/w/cpp/language/types), `int` can only be a 16-bit type on LP32 systems, e.g., Win16 or, maybe, some microcontrollers.  We have no evidence that the library was ever ported to such a system.  If software hasn't been ported, it's not portable, period.  The code is choke-full of usages of `int` so there is no point in an illusory attempt at improving it by using `int32_t` in a handful of places.

If someone wanted to port the library to Win16 that might be an interesting exercise, but that's not what this PR does.

Reply:
++pleroy. @Maratyszcza I maintain an STL, and I honestly I don't think we even worry about 16 bit integers. Let's go with the Google style guide and assume that int is at least 32 bits.